frontend/packages/topology/src/layouts/ColaLayout.ts

```diff
@@ -2,54 +2,105 @@ import * as _ from 'lodash';
 import * as webcola from 'webcola';
 import * as d3 from 'd3';
 import { action } from 'mobx';
-import { Edge, GraphElement, Graph, Layout, Node } from '../types';
+import {
+  ADD_ELEMENT_EVENT,
+  Edge,
+  Graph,
+  GraphElement,
+  isNode,
+  Layout,
+  Node,
+  REMOVE_ELEMENT_EVENT,
+} from '../types';
 import { leafNodeElements, groupNodeElements } from '../utils/element-utils';
-import BaseEdge from '../elements/BaseEdge';
+import {
+  DRAG_MOVE_OPERATION,
+  DRAG_NODE_END_EVENT,
+  DRAG_NODE_START_EVENT,
+  DragEvent,
+  DragNodeEventListener,
+} from '../behavior';
+import { BaseEdge } from '../elements';
 
 class ColaNode implements webcola.Node {
-  private node: Node;
-
-  private nodeIndexx: number;
+  private readonly node: Node;
 
   private xx?: number;
 
   private yy?: number;
 
-  constructor(node: Node, index: number) {
+  private nodeFixed: boolean = false;
+
+  private indexx: number;
+
+  private nodeParent: ColaGroup;
+
+  private distance: number;
+
+  constructor(node: Node, index: number, distance: number) {
     this.node = node;
-    this.nodeIndexx = index;
+    this.indexx = index;
+    this.distance = distance;
   }
 
   get element(): Node {
     return this.node;
   }
 
-  set nodeIndex(newIndex: number) {
-    this.nodeIndexx = newIndex;
+  get id(): string {
+    return this.node.getId();
   }
 
-  get nodeIndex(): number {
-    return this.nodeIndexx;
+  get index(): number {
+    return this.indexx;
   }
 
-  get id(): string {
-    return this.node.getId();
+  set index(index: number) {
+    this.indexx = index;
+  }
+
+  get parent(): ColaGroup {
+    return this.nodeParent;
+  }
+
+  set parent(parent: ColaGroup) {
+    this.nodeParent = parent;
```

Comment:
Doesn't look like you need getters and setters. You can just use the property as is. This would also avoid having names like `indexx`.

There's a bunch of properties like this elsewhere.